File: backend/src/modules/auth/auth.service.ts
import { Injectable } from "@nestjs/common";

@Injectable()
export class AuthService {
  stub(action: string) {
    const responses: Record<string, unknown> = {
      requestOtp: {'status': 'sent', 'ttl_sec': 180},
      verifyOtp: {'status': 'verified', 'token': 'mock_access_token', 'refresh_token': 'mock_refresh_token', 'user': {'id': 'user_001', 'phone': '+233240000000', 'name': 'Yaw Mensah'}},
      refreshToken: {'status': 'refreshed', 'token': 'mock_access_token', 'refresh_token': 'mock_refresh_token'},
    };
    const response = responses[action];
    if (response) {
      return response;
    }
    return { module: "auth", action, status: "ok" };
  }
}

File: mobile/lib/services/auth_service.dart
import "../config/api_endpoints.dart";
import "../app_config.dart";
import "api_client.dart";

class AuthService {
  AuthService(this._client);

  final ApiClient _client;

  Future<Map<String, dynamic>> requestOtp(String phone) async {
    if (AppConfig.useMock) {
      await Future.delayed(const Duration(milliseconds: AppConfig.mockDelayMs));
      return {"status": "sent", "ttl_sec": 180};
    }
    return _client.postJson(ApiEndpoints.otpRequest, {"phone": phone});
  }

  Future<Map<String, dynamic>> verifyOtp(String phone, String code) async {
    if (AppConfig.useMock) {
      await Future.delayed(const Duration(milliseconds: AppConfig.mockDelayMs));
      return {
        "status": "verified",
        "token": "mock_access_token",
        "refresh_token": "mock_refresh_token",
        "user": {"id": "user_001", "phone": phone, "name": "Yaw Mensah"},
      };
    }
    return _client.postJson(ApiEndpoints.otpVerify, {"phone": phone, "code": code});
  }
}
